src/AspNet.Security.OpenId/OpenIdAuthenticationInitializer.cs

```diff
@@ -17,21 +17,14 @@ namespace Microsoft.Extensions.DependencyInjection;
 /// Contains the methods required to ensure that the configuration used by
 /// the OpenID 2.0 generic handler is in a consistent and valid state.
 /// </summary>
+/// <remarks>
+/// Creates a new instance of the <see cref="OpenIdAuthenticationInitializer{TOptions, THandler}"/> class.
+/// </remarks>
 [EditorBrowsable(EditorBrowsableState.Never)]
-public class OpenIdAuthenticationInitializer<TOptions, THandler> : IPostConfigureOptions<TOptions>
+public class OpenIdAuthenticationInitializer<TOptions, THandler>([NotNull] IDataProtectionProvider dataProtectionProvider) : IPostConfigureOptions<TOptions>
     where TOptions : OpenIdAuthenticationOptions, new()
     where THandler : OpenIdAuthenticationHandler<TOptions>
 {
-    private readonly IDataProtectionProvider _dataProtectionProvider;
-
-    /// <summary>
-    /// Creates a new instance of the <see cref="OpenIdAuthenticationInitializer{TOptions, THandler}"/> class.
-    /// </summary>
-    public OpenIdAuthenticationInitializer([NotNull] IDataProtectionProvider dataProtectionProvider)
-    {
-        _dataProtectionProvider = dataProtectionProvider;
```

Comment:
FWIW, I personally tend to null-check all the arguments used in the "DI constructors", so I'd probably keep the field but with a null check (`private readonly IDataProtectionProvider _dataProtectionProvider = dataProtectionProvider ?? throw new ArgumentNullException(dataProtectionProvider);`).

That said, I don't see arguments of DI constructors guarded anywhere in the aspnet-contrib, so... your call 😄

Reply:
I'll leave it as it is as it hasn't changed the existing behaviour and no-one's complained about it before.